nemo/collections/asr/modules/rnnt.py

```diff
@@ -1356,7 +1359,15 @@ def forward(
             # Collect sub batch loss results
             if losses is not None:
                 losses = torch.cat(losses, 0)
-                losses = losses.mean()  # global batch size average
+                target_lengths = torch.cat(target_lengths, 0)
```

Comment:
Remove all this. The RNNT loss calculates the average loss per minibatch provided to it, this mean is just to average over sub-batches.

Reply:
To the best of my understanding, the RNNT Joint [does disable ](https://github.com/NVIDIA/NeMo/blob/61ddd23eba8d88cc08d2f2f7909b75f5333cd4d1/nemo/collections/asr/modules/rnnt.py#L1330)the underlying loss reduction so that it can do its own calculation

Reply:
No it's not. It calls the RNNTLoss which calls NumbaRNNTLoss. Those two are involved for the reductions as you have implemented in this pr. The average step in the joint is only to average these subbatch losses which have already been r reduced by the loss already.
So this file needs to be reverted

Reply:
I should be clearer - the reductions we define by mean batch and mean volume utilise self.loss._reduction and not self.lpss.reduction. This arg is used only for the sum and mean types of reduction, which we remove cause it would do incorrect mean over sub-batches due to super call. 

So RNNT loss has no need to control the reductoon RNNTLoss handles it, joint just averaged losses over sub-batches

Reply:
I just took another pass at the code, with debug statements. I see that self.loss._reduction in RNNTLossNumba gets set to none.

Reply:
Seems to be silent bug for all cases != mean_batch. Good fix !